From d25b692bbf119713fc23dee8e3e1751f415272be Mon Sep 17 00:00:00 2001
From: Raul E Rangel <rrangel@chromium.org>
Date: Wed, 9 Nov 2022 21:26:18 -0700
Subject: [PATCH 2/2] bin/phase-functions: Move du stats into subshell

This avoids polluting the environment variable.

Apparently this calculation isn't hermetic. I'm not sure why:

@@ -268,10 +268,10 @@
 declare -x cros_setup_hooks_run="booya"
 declare -a exclude_hermetic=([0]="--exclude-non-hermetic")
 declare -- f
-declare -a isz=([0]="264" [1]="/var/tmp/portage/dev-libs/libffi-3.1-r8/image/")
-declare -a nsz=([0]="2803" [1]="/var/tmp/portage/dev-libs/libffi-3.1-r8/work")
+declare -a isz=([0]="16" [1]="/var/tmp/portage/dev-libs/libffi-3.1-r8/image/")
+declare -a nsz=([0]="2599" [1]="/var/tmp/portage/dev-libs/libffi-3.1-r8/work")
 declare -- phase_func
 __eapi6_src_install ()

Change-Id: I3970e01d6955f17835f2db205e44258d661a4902
---
 bin/phase-functions.sh | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/usr/lib/portage/PYTHON_VERSION/phase-functions.sh b/usr/lib/portage/PYTHON_VERSION/phase-functions.sh
index 0a0c3037a..5f3b4893b 100644
--- a/usr/lib/portage/PYTHON_VERSION/phase-functions.sh
+++ b/usr/lib/portage/PYTHON_VERSION/phase-functions.sh
@@ -619,12 +619,11 @@ __dyn_install() {
 
 	# record build & installed size in build log
 	if type -P du &>/dev/null; then
-		local nsz=( $(du -ks "${WORKDIR}") )
-		local isz=( $(du -ks "${D}") )
-
 		# subshell to avoid polluting the caller env with the helper
 		# functions below
 		(
+			local nsz=( $(du -ks "${WORKDIR}") )
+			local isz=( $(du -ks "${D}") )
 			# align $1 to the right to the width of the widest of $1 and $2
 			padl() {
 				local s1=$1
-- 
2.38.1.431.g37b22c650d-goog

